### PR TITLE
Restructure + nodenext compatibility

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Fdawgs — APPROVED** [review]
LGTM regarding general changes.

No idea about the TypeScript changes though!

**Comment by Uzlopak** [comment]
@Fdawgs 
Thats oK. Thats why I also requested a review by @climba03003 ;)

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+     * Pre-configured instance of MongoClient
+     */
+    client?: MongoClient;
+    /**
+     * Connection url
+     */
+    url?: string;
+  }
+
+  export class ObjectId extends MongodbObjectId {}
```
Comment (climba03003) [inline types/index.d.ts:55]: Why not `export { ObjectId }` like before?

Comment (Uzlopak) [reply types/index.d.ts]: done

**Comment by Uzlopak** [comment]
@climba03003 PTAL

**Review by Eomm — APPROVED** [review]

**Review by climba03003 — APPROVED** [review]

**Pull request merged**